### PR TITLE
Account for delegate creation inside an explicitly-converted tuple expression

### PR DESCRIPTION
**Pull request by 333fred**
Fixes https://github.com/dotnet/roslyn/issues/64774.

### CODE REVIEW

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for review.

File: src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IDelegateCreationExpression.cs [hunk 1]
```diff
@@ -1998,5 +1998,37 @@ void M(Action a1, Action a2, Action a3)
 
             VerifyFlowGraphAndDiagnosticsForTest<BlockSyntax>(source, expectedFlowGraph, expectedDiagnostics);
         }
+
+        [Fact, WorkItem(64774, "https://github.com/dotnet/roslyn/issues/64774")]
+        public void ExplicitCastOnTuple()
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IDelegateCreationExpression.cs:2003]: >ExplicitCastOnTuple [](http://example.com/codeflow?start=20&length=19)

Consider adding a test for VB

File: src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IDelegateCreationExpression.cs [hunk 2]
```diff
@@ -1998,5 +1998,37 @@ void M(Action a1, Action a2, Action a3)
...
+        {
+            var code = """
+                using System;
+
+                (int i, Action invoke) = /*<bind>*/((int, Action))(1, M)/*</bind>*/;
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IDelegateCreationExpression.cs:2008]: >Action [](http://example.com/codeflow?start=24&length=6)

Consider adding a test for a function pointer type as well.

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 1), additional tests are suggested

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for a second review.

**Review by chsienki — APPROVED** [review]

**Pull request merged**